src/wasm/literal.cpp

```diff
@@ -2352,41 +2352,41 @@ Literal Literal::narrowUToVecI16x8(const Literal& other) const {
 
 enum class LaneOrder { Low, High };
 
-template<size_t Lanes,
-         LaneArray<Lanes * 2> (Literal::*IntoLanes)() const,
-         LaneOrder Side>
+template<size_t Lanes, typename LaneFrom, typename LaneTo, LaneOrder Side>
 Literal extend(const Literal& vec) {
-  LaneArray<Lanes* 2> lanes = (vec.*IntoLanes)();
+  LaneArray<Lanes* 2> lanes = getLanes<LaneFrom, Lanes * 2>(vec);
   LaneArray<Lanes> result;
   for (size_t i = 0; i < Lanes; ++i) {
-    result[i] = lanes[(Side == LaneOrder::Low) ? i : i + Lanes];
+    result[i] =
+      Literal((LaneTo)(LaneFrom)lanes[(Side == LaneOrder::Low) ? i : i + Lanes]
+                .geti32());
```

Comment:
It might be worth have one or two helper variables in here just to break it up and make it more readable.